### ISSUE TITLE
Do not fail tests based on Timeout when debugging

### ISSUE THREAD

**Issue by Tyrrrz** [microsoft/testfx#291]
## Description
It would be nice if the tests didn't fail due to timeout while debugging, just like in NUnit. Currently, you need to remove the attribute to properly debug such tests, because otherwise it's aborted abruptly in the middle of debug.

## Steps to reproduce
- Put `[Timeout]` attribute on a test method.
- Launch the test method in debug mode and enter some breakpoint
- Wait until the timeout is exceeded and then step over

## Expected behavior
- Debugging should continue without problems because the timeout is disabled during debug

## Actual behavior
- The test has stopped abruptly, not letting you finish debugging

## Environment
Windows 10
vstest.console v15.0.26929.2
nuget packages v1.2.0

**Comment by AbhitejJohn** [comment]
I think that's a fair point. This could be really annoying.

**Comment by dotMorten** [comment]
I agree it's super annoying, however how do you debug a testrun where some tests are actually timing out? Ie you do a test run with the debugger attached, a test run could hang before you get to the one you were interested in debugging

**Comment by Kissaki** [comment]
Do you want/expect it to time only while the application is not paused, or never timeout indefinitely?

I would fine a lack of timeout surprising and unexpected when the test specifically has a timeout specified.

**Comment by nohwnd** [comment]
Link added to the list.

**Comment by dotMorten** [comment]
I don't think just having the debugger attached should define a timeout. If you do a full test run with the debugger attached, you could have a test hang indefinitely. IMHO it should only be pausing while you're at a breakpoint, but I don't quite think that's possible to do without some ugly timer loop that also gets paused instead of relying on timestamps.

**Comment by nohwnd** [comment]
I agree with you dotMorten, having the timer pause when the application is paused seems like the best solution, that way I can debug the code and still have it behave close to what it would behave like in real life.
